sample-tracking-status-overview-app/src/main/groovy/life/qbic/portal/sampletracking/datasources/samples/SamplesDbConnector.groovy

```diff
@@ -0,0 +1,106 @@
+import java.sql.ResultSet
+
+/**
+ * <b>Database connector to the sample tracking database</b>
+ *
+ * <p>This connector provides functionality to receive sample information for samples tracked by
+ * the sample tracking system that is in place at QBiC.</p>
+ * <p>Furthermore the connector implements the {@link CountSamplesDataSource} and provides information
+ * that is required by the respective use case</p>
+ *
+ * @since 1.0.0
+ */
+@Log4j2
+class SamplesDbConnector implements CountSamplesDataSource {
+    private final ConnectionProvider connectionProvider
+
+    /**
+     * Creates a database connector that will connect to the database using the provided connection
+     * provider.
+     * @param connectionProvider the connection provider providing the connections that are used to
+     * connect to the database
+     * @since 1.0.0
+     */
+    SamplesDbConnector(ConnectionProvider connectionProvider) {
+        this.connectionProvider = connectionProvider
+    }
+
+
+    /**
+     * {@inheritDoc}
+     * <p><b><i>PLEASE NOTE: In case multiple statuses were entered at the same time,
+     * this method adds all statuses to the returned list!</i></b></p>
+     * @since 1.0.0
+     */
+    @Override
+    List<Status> fetchSampleStatusesForProject(String projectCode) throws DataSourceException {
+        String queryTemplate = constructQuery()
+        Connection connection = connectionProvider.connect()
+        List<Status> statuses = new ArrayList<>()
+        String sqlRegex = "$projectCode%"
+        connection.withCloseable {
+            PreparedStatement preparedStatement = it.prepareStatement(queryTemplate)
+            preparedStatement.setString(1, sqlRegex)
+            ResultSet resultSet = preparedStatement.executeQuery()
+            while (resultSet.next()) {
+                String sampleCode = resultSet.getString("sample_id")
+                String sampleStatusString = resultSet.getString("sample_status")
+                String arrivalTime = resultSet.getString("arrival_time")
+                Status sampleStatus
+                try {
+                    sampleStatus = Status.valueOf(sampleStatusString)
+                } catch(IllegalArgumentException statusNotFound) {
+                    // The status in the database is invalid. This should never be the case!
+                    log.error("Could not parse status $sampleStatusString for $sampleCode at $arrivalTime", statusNotFound)
+                    throw new DataSourceException("Retrieval of sample statuses failed for sample $sampleCode")
+                }
+                statuses.add(sampleStatus)
+            }
+        }
+        return statuses
+    }
+
+    static String constructQuery() {
+        /*
+        The filter criteria to avoid applying the query to the whole table.
+        Replace `?` with your matching sample_id.
+         */
+        final String filterCriteria = "WHERE sample_id LIKE ? AND sample_id NOT LIKE \"%ENTITY%\""
```

Comment:
Is it intended that you have it like this, I would assume it must be:

```suggestion
        final String filterCriteria = "WHERE sample_id LIKE ? AND sample_id NOT LIKE \%ENTITY%\"
```

Reply:
Nope we need to escape the `""` chars.